cli/pkg/update/update.go

```diff
@@ -0,0 +1,162 @@
+package update
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/hashicorp/go-version"
+	"github.com/rilldata/rill/cli/pkg/dotrill"
+)
+
+const addr = "https://api.github.com/repos/rilldata/rill-developer/releases/latest"
+
+func CheckVersion(ctx context.Context, currentVersion string) error {
+	// Check if build from source
+	if currentVersion == "" {
+		return nil
+	}
+
+	latestVersion, err := checkVersion(ctx)
+	if err != nil {
+		return err
+	}
+
+	v1, err := version.NewVersion(currentVersion)
+	if err != nil {
+		return err
+	}
+
+	v2, err := version.NewVersion(latestVersion)
+	if err != nil {
+		return err
+	}
+
+	if v1.LessThan(v2) {
+		fmt.Printf("\n%s %s → %s\n\n",
+			color.YellowString("A new version of rill is available:"),
+			color.CyanString(currentVersion),
+			color.CyanString(latestVersion))
+		return nil
+	}
+
+	return nil
+}
+
+func checkVersion(ctx context.Context) (string, error) {
+	latestVersion, err := LatestVersion(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	err = dotrill.SetVersionUpdatedAt(time.Now().Format("2006-01-02 15:04"))
+	if err != nil {
+		return "", err
+	}
+
+	err = dotrill.SetVersion(latestVersion)
+	if err != nil {
+		return "", err
+	}
+
+	return latestVersion, nil
+}
+
+// This will return the latest version available in cache or fetch it from github if its older than 24h
+func LatestVersion(ctx context.Context) (string, error) {
+	cachedVersion, err := dotrill.GetVersion()
+	if err != nil {
+		return "", err
+	}
+
+	if cachedVersion != "" {
+		cachedVersionUpdatedAt, err := dotrill.GetVersionUpdatedAt()
+		if err != nil {
+			return "", err
+		}
+
+		updatedAt, err := time.Parse("2006-01-02 15:04", cachedVersionUpdatedAt)
+		if err != nil {
+			return "", err
+		}
+
+		if time.Since(updatedAt).Hours() > 24 {
+			// Check with latest release on github
+			info, err := fetchLatestVersion(ctx)
+			if err != nil {
+				return "", err
+			}
+
+			return info.Version, nil
+		}
+
+		return cachedVersion, nil
+	}
```

Comment:
1. Change this condition. If `time.Since` is less than the TTL, return the cached version. Else, don't return anything, so the fetch on line 102 is triggered.
2. Also, make the TTL a const at the top of the file – like `const versionCheckTTL = 24*time.Hour`